pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTEnumDeclaration.java

```diff
@@ -18,4 +21,29 @@ public Object jjtAccept(JavaParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
 
+    public boolean isNested() {
```

Comment:
Yes it would probably make sense to move `isNested` to the interface. However merging `makeOuterClassOf` and `makeNestedClassOf` would probably make the recursive call messier for little benefit :/